Contuity/Models/Jot.swift

```diff
@@ -66,12 +66,95 @@ extension Jot: DatabaseProtocol {
     func write() {
         let insert = "INSERT INTO jot (id, data, queue, createdAt, modifiedAt, latitude, longitude)"
         let values =
-        "VALUES (\(id), \"\(data ?? "")\", \(queue), \"\(createdAt)\", \"\(createdAt)\", \(latitude ?? 0), \(longitude ?? 0))"
+        "VALUES (\(id), \"\(data)\", \(queue), \"\(createdAt)\", \"\(modifiedAt ?? createdAt)\", \(latitude ?? 0), \(longitude ?? 0))"
 
         guard let conn = DatabaseManager.shared.conn,
             let statement = try? conn.prepare("\(insert) \(values)") else {
                 return
         }
         _ = try? statement.run()
     }
+    
+    func update() {
+        let dataMessage = "data = \"\(data)\""
+        let queueMessage = "queue = \(queue)"
+        let createdMessage = "createdAt = \"\(createdAt)\""
+        let modifiedMessage = "modifiedAt = \"\(modifiedAt ?? createdAt)\""
+        let latMessage = "latitude = \(latitude ?? 0)"
+        let longMessage = "longitude = \(longitude ?? 0)"
+        let setMessage = "\(dataMessage), \(queueMessage), \(createdMessage), \(modifiedMessage), \(latMessage), \(longMessage)"
+        let update = "UPDATE jot SET \(setMessage) WHERE id = \(id)"
+        
+        guard let conn = DatabaseManager.shared.conn,
+            let statement = try? conn.prepare(update) else {
+                return
+        }
+        _ = try? statement.run()
+    }
+    
+    static func read(givenID: Int) -> Jot? {
+        guard let conn = DatabaseManager.shared.conn
+        else {
+            print("conn")
+            return nil
+        }
+        do {
+            for row in try conn.prepare("SELECT * FROM jot WHERE id = \(givenID)") {
```

Comment:
I think all this logic can be condensed to a smaller function

Reply:
If you're interested I did it in my branch:
`    static func read(givenID: Int) throws -> Jot {
        guard let conn = DatabaseManager.shared.conn
            else {
                throw DatabaseError.selectFailed
        }
        do {
            for row in try conn.prepare("SELECT * FROM jot WHERE id = \(givenID)") {
                return try parseRow(row: row)
            }
        }
        catch {
            throw DatabaseError.selectFailed
        }
        throw DatabaseError.selectFailed
    }
    
    
    /// Parses a single row from the database into a Jot
    ///
    /// - Parameter row: a row from the database
    /// - Returns: a Jot
    /// - Throws: DatabaseError.selectFailed if the database SELECT failed
    static func parseRow(row: Statement.Element) throws -> Jot {
        guard let optionalID: Int64 = row[0] as? Int64,
            let newData = row[1] as? String
            else {
                throw DatabaseError.selectFailed
        }
        let id = Int(optionalID)
        let row2 = row[2]
        var newQueue = false
        switch row2 {
        case let row2 as Int64:
            if row2 == 1 {
                newQueue = true
            }
        default:
            break
        }
        var newCreatedAt = ""
        let row3 = row[3]
        switch row3 {
        case let row3 as String:
            newCreatedAt = String(row3)
        default:
            throw DatabaseError.selectFailed
        }
        guard let newModifiedAt = row[4] as? String? else{
            throw DatabaseError.selectFailed
        }
        var newLat: Double?
        let row5 = row[5]
        switch row5 {
        case let row5 as Double:
            newLat = Double(row5)
        default:
            throw DatabaseError.selectFailed
        }
        var newLong: Double?
        let row6 = row[6]
        switch row6 {
        case let row6 as Double:
            newLong = Double(row6)
        default:
            throw DatabaseError.selectFailed
        }
        return Jot(id: id,
                   data: newData,
                   queue: newQueue,
                   createdAt: newCreatedAt,
                   modifiedAt: newModifiedAt,
                   latitude: newLat,
                   longitude: newLong)
    }`
We can probably just leave it like this and I'll just do the refactor in my branch

Reply:
sgtm